components/aws/sagemaker/batch_transform/README.md

```diff
@@ -0,0 +1,98 @@
+environment | The environment variables to set in the Docker container | Yes | Yes | Dict | Maximum length of 1024. Key Pattern: `[a-zA-Z_][a-zA-Z0-9_]*`. Value Pattern: `[\S\s]*`. Upto 16 key and values entries in the map | |
+
+The following parameters construct [TransformInput](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformInput.html) object of the input. These describe the input source and the way the transform job consumes it.
+
+Argument        | Description                 | Optional (in pipeline definition) | Optional (in UI) | Data type  | Accepted values | Default    |
+:---            | :----------                 | :----------                       | :----------      | :----------| :----------     | :----------|
+input_location | The S3 location of the data source that is associated with a channel. [Read more on S3Uri](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformS3DataSource.html) | No | No | String | | |
+data_type | Used by Sagemaker to identify the objects from the S3 bucket to be used for batch transform. [Read more on S3DataType](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformS3DataSource.html) | Yes | Yes | String | `ManifestFile`, `S3Prefix`, `AugmentedManifestFile`| S3Prefix |
+content_type | The multipurpose internet mail extension (MIME) type of the data. Amazon SageMaker uses the MIME type with each http call to transfer data to the transform job | Yes | Yes | String | | |
+split_type | The method to use to split the transform job data files into smaller batches | Yes | Yes | String | `Line`, `RecordIO`, `TFRecord`, `None` | None |
+compression_type | If the transform data is compressed, specify the compression type | Yes | Yes | String | `GZip`, `None` | None |
+
+* input_location and data_type parameters above are used to construct [S3DataSource](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformS3DataSource.html) object which is part of [TransformDataSource](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformDataSource.html) object in [TransformInput](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformInput.html) part of the batch transform API.
+```
+TransformInput={
+        'DataSource': {
+            'S3DataSource': {
+                'S3DataType': 'ManifestFile'|'S3Prefix'|'AugmentedManifestFile',
+                'S3Uri': 'string'
+            }
+        },
+        'ContentType': 'string',
+        'CompressionType': 'None'|'Gzip',
+        'SplitType': 'None'|'Line'|'RecordIO'|'TFRecord'
+    }
+```
+[Ref](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_transform_job)
+
+The following parameters construct [TransformOutput](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformOutput.html) object of the input. These describe the results of a transform job.
```

Comment:
This first sentence doesn't seem super clear to me. What's an "object of the input"? An input parameter?

Reply:
not a parameter, I mean to say these set of input parameters collectively represent this input object (e.g. TransformOutput here) in the API. since these are represented as dictionaries in the API, I found this word appropriate and is also used in aws docs, let me know if it makes sense or if you have any other suggestions

Reply:
Changed it to say - *The following parameters are used to construct `XYZ` object of the CreateTransformJob API*